Assets/Scripts/InventoryAndItemsScripts/InventoryController.cs

```diff
@@ -0,0 +1,227 @@
+        }
+    }
+
+    void NavInvLeft(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex > 0 && inventoryIndex % 6 != 0) 
+            {
+                inventoryIndex--;
+                selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+                selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+                Debug.Log(inventoryIndex);
+            }
+        }
+    }
+
+    void NavInvRight(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex < inventorySlots.Count && inventoryIndex % 6 != 5)
+            {
+                inventoryIndex++;
+                selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+                selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+                Debug.Log(inventoryIndex);
+            }
+        }
+    }
+
+    void NavInvUp(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex > 5) inventoryIndex -= 6;
+            selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+            selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+            Debug.Log(inventoryIndex);
+        }
+    }
+
+    void NavInvDown(CallbackContext c)
+    {
+        if (inventory.activeSelf)
+        {
+            if (inventoryIndex < inventorySlots.Count - 6) inventoryIndex += 6;
+            selectedSlot.transform.SetParent(invPanel.transform.GetChild(inventoryIndex));
+            selectedSlot.transform.position = invPanel.transform.GetChild(inventoryIndex).position;
+            Debug.Log(inventoryIndex);
+        }
+    }
+
+    private void Update()
+    {
+        List<ItemData> keysToRemove = new List<ItemData>();
+        foreach (var item in pInv.ItemDictionary)
+        {
+            if (item.Key.itemName.Contains("Coin"))
+            {
+                int coinCount = int.Parse(coins.transform.GetChild(1).gameObject.GetComponent<TextMeshProUGUI>().text);
+                coins.transform.GetChild(1).gameObject.GetComponent<TextMeshProUGUI>().text = (coinCount + 1).ToString();
+                keysToRemove.Add(item.Key);
+            } else if (!itemsInInventory.Contains(item.Key))
+            {
+                itemsInInventory.Add(item.Key);
+                var slot = inventorySlots[itemsInInventory.Count-1];
+                var img = item.Key.sprite;
+                slot.GetComponent<UnityEngine.UI.Image>().sprite = img;
+            }
+        }
+
+        // Remove keys
+        foreach (var key in keysToRemove)
+        {
+            pInv.ItemDictionary.Remove(key);
+        }
+
+        var title = itemDesc.transform.GetChild(0).gameObject;
+        var desc = itemDesc.transform.GetChild(1).gameObject;
```

Comment:
you should definitely tag the children and get them by their tags, since the order might get fucked with when we update things